src/Symfony/Component/DependencyInjection/Compiler/ServiceReferenceGraph.php

```diff
@@ -80,17 +69,12 @@ public function clear()
 
     /**
      * Connects 2 nodes together in the Graph.
-     *
-     * @param string $sourceId
-     * @param string $sourceValue
-     * @param string $destId
-     * @param string $destValue
-     * @param string $reference
-     * @param bool   $lazy
-     * @param bool   $weak
      */
-    public function connect($sourceId, $sourceValue, $destId, $destValue = null, $reference = null, bool $lazy = false, bool $weak = false)
+    public function connect(?string $sourceId, $sourceValue, ?string $destId, $destValue = null, string $reference = null, bool $lazy = false, bool $weak = false)
```

Comment:
Now that we are removing the phpdoc, some arguments in [this thread](https://github.com/symfony/symfony/issues/18890) are becoming invalid. Consider reopening it. Also, I failed to mention in the thread that doing this is a recipe for git conflicts, which surely you can confirm, I bet most conflicts you experience when merging are about method signatures.

Fabien said:

> Indeed, we are not using multiline method signature, that's the choice we made and we won't change it as it would make merging branches a nightmare for no benefits.

But now I think switching to multiline method signatures would make merging branches way easier after that one-shot nightmare.

Reply:
I don't agree :) I value the vertical space in my IDE, I certainly don't want to have it polluted by multi line signatures.

Reply:
@nicolas-grekas you're implying signatures have no value, but the 7.0 type hinting might change your view a lot, because now signatures are really helfpul to figure out what you are dealing with.

Also, why not add `string` type hinting to `sourceValue` and `destValue`? If it is because they can actually be something else, it should be documented right. I only noticed it now, single line signatures make it hard to notice details like this.

> I value the vertical space in my IDE

Consider using split windows if you don't, that will allow to display way more code on your screen, but it comes with one condition though: avoid long lines.

Reply:
For the record, I'm also against splitting signature on multiple lines.

Reply:
😢

Reply:
I wrote a whole blog post about this just now, in case you have some time to kill: https://dev.to/greg0ire/the-case-for-multiline-signatures-dl6